pkg/kubectl/cmd/util/factory.go

```diff
@@ -165,6 +166,24 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 			if cfg.GroupVersion != nil {
 				cmdApiVersion = *cfg.GroupVersion
 			}
+			client, err := clients.ClientForVersion(&unversioned.GroupVersion{Version: "v1"})
+			if err != nil {
+				fmt.Printf("Failed to create client: %v", err)
+				return nil, nil
+			}
+			list, err := client.ThirdPartyResources(api.NamespaceDefault).List(api.ListOptions{})
```

Comment:
> Shouldn't we negotiate the client with clients.ClientForVersion(&unversioned.GroupVersion{Group: "extensions"}) then?

Currently the version negotiation is a mess. We hard code the client version anyway (https://github.com/kubernetes/kubernetes/blob/master/pkg/client/unversioned/helper.go#L371).

I agree we should pass "extensions/v1beta1" here if it doesn't cause any error, then we don't need to fix this code when we fix the client.version for real.